src/components/DonutPath.js

```diff
@@ -0,0 +1,65 @@
+const React = require('react');
+const d3 = require('d3');
+
+const DonutPath = React.createClass({
+  propTypes: {
+    arc: React.PropTypes.func,
+    color: React.PropTypes.string,
+    opacity: React.PropTypes.number,
+    value: React.PropTypes.object
+  },
+
+  getInitialState () {
+    return {
+      elapsedTime: 0,
+      endAngle: 0
+    };
+  },
+
+  componentDidMount () {
+    if (this.props.animateOnLoad) {
```

Comment:
Question on this. I am passing all the props via {...this.props}. Do I need to repeat all the propTypes, too?